### PR TITLE
Bug fixing 4 and test cases

### LINKED ISSUE #209: [PE-D] edit command requires addtional input not specific in the User Guide
Here is how I got to this view:
1) Start the app
2) Input "add add n/Blair project dl/29-02-2020 00:00:00 ru/http://github.com/a/b.git d/Coding in Greenwich tg/challenging"
3) Input edit "edit n/Blair project dl/01-02-2020 00:00:00 ru/http://github.com/a/b.git d/Coding in Greenwich tg/challenging" Changinging only the deadline of the project

The error message pictured in the picture says an index is required. Yet the User Guide does not mention this at all. The example provided by the User Guide does not show this as well. (see screenshot of user guide)

Do make the changes in your User Guide

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: euzintan/ped#3

### LINKED ISSUE #224: [PE-D] Typo in viewteammate documentation
The example given is:
```Example: viewteammate Lucas98 displays all the information about the teammate with the Git User Name Lucas 98 to the user.```

The space between Lucas and 98 is misleading. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: jazerler/ped#13

### LINKED ISSUE #237: [PE-D] Documentation edit command not working
Missing index in edit function

Error message says : 
Invalid command format! 
edit: Edits the details of the project identified by the index number used in the displayed project list. Existing values will be overwritten by the input values.
Parameters: INDEX (must be a positive integer) [n/PROJECTNAME] [dl/DEADLINE] [ru/REPOURL] [d/PROJECTDESCRIPTION] [tg/PROJECT TAG]...
[tk/TASK]...
Example: edit 1 dl/29-02-2020 00:00:00 ru/https://github.com/a/a.git

Need to use index

-------------
Labels: `severity.High` `type.DocumentationBug`
original: junlong4321/ped#8

### LINKED ISSUE #239: [PE-D] Extra ` in exit
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: junlong4321/ped#10

### LINKED ISSUE #240: [PE-D] Inconsistent full stops
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: junlong4321/ped#11

### LINKED ISSUE #241: [PE-D] Grammatical error
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: junlong4321/ped#12

### LINKED ISSUE #243: [PE-D] Missing hyphen
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: junlong4321/ped#14

### LINKED ISSUE #250: [PE-D] Edit Project Section Inaccuracy In User Guide
Currently there is a small issue with the documentation for the edit project section. As mentioned early on in the user guide, parameters in square brackets are optional. 

However, commands such as `edit` and `edit n/Charletan` were rejected. The `edit n/Charletan` example is shown below:

While the commands I tried were obviously rejected for good reason, it would seem that there were some parameters that were not optional, which is not expressed in the edit project section.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: IronBiscuit/ped#6

### LINKED ISSUE #287: Participation not updated when project name is changed as project is found through the name

### PR DESCRIPTION
**Pull request by lucastai98**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ Example: `edit 1 n/Resident Evil project /d new horror` changes the name of the
 
 # **Features** in project scope
 
-## **Task** related features
+## **Task**-related features
```
Comment (TCQian) [inline docs/UserGuide.md]: make the "Task-related features" bold.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -332,7 +332,7 @@ Format: `viewtask INDEX `
 
 Example: `viewtask 4` displays all information from task number 4 in the list.
 
-## **Teammate** related features
+## **Teammate**-related features
```
Comment (TCQian) [inline docs/UserGuide.md]: make the "Teammate-related features" bold

File: docs/UserGuide.md [hunk 3]
```diff
@@ -206,6 +206,7 @@ Format: `find [KEYWORD...]`
 - The search is case-insensitive. e.g run will match Run.
 - Only the name of the projects are searched.
 - Can be multiple words.
+- Keywords provided must be complete words and will only match complete words from the name of the project.
```
Comment (TCQian) [inline docs/UserGuide.md]: complete word or words?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -290,10 +290,11 @@ Filters tasks in the task list by various predicates:
 Format: `filter (ta/TASK_ASSIGNEE_NAME)||(td/DEADLINE)||(start/START_DATE end/END_DATE)||(tp/TASK PROGRESS)||(tn/TASK_NAME)` 
   - User may choose one predicate to filter tasks by
   - Assignee name is the name of the Teammate who is assigned to the task
+  - Task Name can be any alphanumeric value (containing only alphabets and / or numbers).
+  - Names provided must be complete words and will only match complete words from the name of the task or assignee.
```
Comment (TCQian) [inline docs/UserGuide.md]: names or name? word or words?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -153,30 +153,32 @@ Example: `listpersons` lists all persons in the catalogue to the user.
...
 
 ### Start work on an existing project `startproject `
+
 Initialises the project specified.
 
 Format: `startproject (INDEX)`
-- Initialises the project at the specified INDEX.
-- The index refers to the index number shown in the displayed project list.
+- Initialises the project at the specified INDEX
+- The index refers to the index number shown in the displayed project list
```
Comment (TCQian) [inline docs/UserGuide.md:167]: Could you please clarify which point should contains full stop and which should not? Why "List all persons....catalogue." has full stop but then "Leaves the current page......scope" has no?

File: src/test/java/seedu/address/logic/parser/AddTaskCommandParserTest.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + PROJECT_NAME_DESC_BOB + DEADLINE_DESC_B
+                + REPOURL_DESC_B
+                + PROJECT_DESCRIPTION_DESC_BOB + PROJECT_TAG_DESC_FRIEND + TASK_DESC_DG, new AddCommand(
+                expectedProject));
+
+        // multiple project names - last name accepted
+        assertParseSuccess(parser, PROJECT_NAME_DESC_AMY + PROJECT_NAME_DESC_BOB + DEADLINE_DESC_B
+                        + REPOURL_DESC_B + PROJECT_DESCRIPTION_DESC_BOB + PROJECT_TAG_DESC_FRIEND + TASK_DESC_DG,
+                new AddCommand(
+                        expectedProject));
```
Comment (TCQian) [inline src/test/java/seedu/address/logic/parser/AddTaskCommandParserTest.java]: Why the line is split into 2? So for the rest.

File: src/test/java/seedu/address/logic/parser/AddTaskCommandParserTest.java [hunk 7]
```diff
@@ -55,157 +58,81 @@
...
-                .withTasks(testTask1, testTask2)
-                .build();
-        assertParseSuccess(parser, PROJECT_NAME_DESC_BOB + DEADLINE_DESC_B + REPOURL_DESC_B
-                        + PROJECT_DESCRIPTION_DESC_BOB
-                        + PROJECT_TAG_DESC_FRIEND + PROJECT_TAG_DESC_HUSBAND + TASK_DESC_DG + TASK_DESC_MODEL,
-                new AddCommand(expectedProjectMultipleTasks));
-    }
+        assertParseSuccess(parser, TASK_A_NAME + TASK_B_DEADLINE+TASK_A_DEADLINE
+                + TASK_A_PROGRESS, new AddTaskCommand(
+                expectedTask));
```
Comment (TCQian) [inline src/test/java/seedu/address/logic/parser/AddTaskCommandParserTest.java]: Why line is split into 2. It seems like it doesn't exceed the limit of a line of code.

File: docs/UserGuide.md (hunk 5 above)
Comment (lucastai98) [reply docs/UserGuide.md:167]: anything in point form does not have full stops, everything else has full stops

File: docs/UserGuide.md (hunk 4 above)
Comment (lucastai98) [reply docs/UserGuide.md]: Yes, names refer to both task name and assignee name. Words as you can have more than 1 keyword

File: docs/UserGuide.md (hunk 3 above)
Comment (lucastai98) [reply docs/UserGuide.md]: Words as you can have more than one keyword

Comment (TCQian) [reply docs/UserGuide.md]: Ok!

File: docs/UserGuide.md (hunk 5 above)
Comment (TCQian) [reply docs/UserGuide.md:167]: Make sense.

**Review by TCQian — APPROVED** [review]
LGTM

**Pull request merged**